### PR TITLE
Implement Array.isArray and Array#copyWithin

### PR DESCRIPTION
**Pull request by MaxGraey**

### CODE REVIEW

File: std/assembly/array.ts [hunk 1]
```diff
@@ -27,11 +27,20 @@ import {
   MAX_DOUBLE_LENGTH
 } from "./internal/number";
 
+@inline
+function mix<T extends number>(a: T, b: T): T {
```
Comment (dcodeIO) [inline std/assembly/array.ts]: To me, it's not quite obvious what this is doing just from its name. Would favor to not use a helper here.

File: std/assembly/array.ts [hunk 2]
```diff
@@ -27,11 +27,20 @@ import {
...
+  return <T>(a < 0 ? max<T>(b + a, 0) : min<T>(a, b));
+}
+
 export class Array<T> {
 
   /* @internal */ buffer_: ArrayBuffer;
   /* @internal */ length_: i32;
 
+  @inline static isArray<U>(value: U): bool {
+    return isArray(value) && value !== null;
```
[the remark this answers is not in the document]
Comment (MaxGraey) [reply std/assembly/array.ts]: Done!

File: std/assembly/array.ts (hunk 1 above)
Comment (MaxGraey) [reply std/assembly/array.ts]: Done!

**Comment by dcodeIO** [comment]
Great, thanks! :)

**Pull request merged**